test/e2e/common/pods.go

```diff
@@ -848,4 +853,113 @@ var _ = framework.KubeDescribe("Pods", func() {
 		validatePodReadiness(false)
 
 	})
+
+	ginkgo.It("should run through the lifecycle of Pods and PodStatus", func() {
+		podResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+		testNs := f.Namespace.Name
+		testPodName := "pod-test"
+		testPodImage := "nginx"
+		testPodImage2 := "httpd"
+		testPodLabels := map[string]string{"test-pod-static": "true"}
+		testPodLabelsFlat := "test-pod-static=true"
+
+		testPod := v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testPodName,
+				Labels: testPodLabels,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  testPodName,
+						Image: testPodImage,
+					},
+				},
+			},
+		}
+		ginkgo.By("creating a Pod with a static label")
+		_, err := f.ClientSet.CoreV1().Pods(testNs).Create(context.TODO(), &testPod, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create Pod")
+
+		ginkgo.By("setting up a watch for the Pod")
+		podWatchTimeoutSeconds := int64(180)
+		podWatch, err := f.ClientSet.CoreV1().Pods(testNs).Watch(context.TODO(), metav1.ListOptions{LabelSelector: testPodLabelsFlat, TimeoutSeconds: &podWatchTimeoutSeconds})
+		framework.ExpectNoError(err, "failed to set up watch")
+
+		podWatchChan := podWatch.ResultChan()
+		ginkgo.By("watching for Pod to be ready")
+		for watchEvent := range podWatchChan {
+			podWatchEvent, ok := watchEvent.Object.(*v1.Pod)
+			framework.ExpectEqual(ok, true, "unable to assert runtime object type to v1.Pod")
+			if podWatchEvent.Status.Phase == v1.PodRunning {
```

Comment:
can this watch channel block indefinitely if the condition is not met and the channel is not closed?
[1]

Reply:
I believe it does block indefinitely until the timeout happens

Reply:
is there a way to print an error and abort the test if that happens?

Reply:
@hh interested if possible to catch the above mentioned error on watch timeout.

Reply:
this also needs to set a flag outside the for loop, needs to set it inside the loop, and do a framework.Expect after the for loop

Reply:
Added checks for watches in this test in 9a8e31bacb3

Reply:
the latest update LGTM, thanks.